EKF/control.cpp

```diff
@@ -518,32 +518,42 @@ void Ekf::controlGpsFusion()
 		// Detect if coming back after significant time without GPS data
 		bool gps_signal_was_lost = isTimedOut(_time_prev_gps_us, 1000000);
 
-		// GPS yaw aiding selection logic
-		if ((_params.fusion_mode & MASK_USE_GPSYAW)
-				&& ISFINITE(_gps_sample_delayed.yaw)
-				&& _control_status.flags.tilt_align
-				&& (!_control_status.flags.gps_yaw || !_control_status.flags.yaw_align)
-				&& !_gps_hgt_intermittent) {
-
-			if (resetGpsAntYaw()) {
-				// flag the yaw as aligned
-				_control_status.flags.yaw_align = true;
 
-				// turn on fusion of external vision yaw measurements and disable all other yaw fusion
-				_control_status.flags.gps_yaw = true;
-				_control_status.flags.ev_yaw = false;
-				_control_status.flags.mag_dec = false;
+		if (!(_params.fusion_mode & MASK_USE_GPSYAW)
+		    || _is_gps_yaw_faulty) {
 
-				stopMagHdgFusion();
-				stopMag3DFusion();
+			stopGpsYawFusion();
+
+		} else {
+			if (ISFINITE(_gps_sample_delayed.yaw)) {
+
+				if (!_control_status.flags.gps_yaw
+				    && _control_status.flags.tilt_align
+				    && !_gps_hgt_intermittent) {
 
-				ECL_INFO_TIMESTAMPED("starting GPS yaw fusion");
+					// Activate GPS yaw fusion
+					if (resetGpsAntYaw()) {
+						_control_status.flags.yaw_align = true;
+
+						startGpsYawFusion();
+
+						ECL_INFO_TIMESTAMPED("starting GPS yaw fusion");
```

Comment:
You could move this print statement into the `startGpsYawFusion` function

Reply:
Annoying reminder to also consider if it's actually needed or could be reflected in the status flags. Printing to STDOUT can potentially impact real time performance.

Reply:
I was thinking of getting rid of it: there is no message to tell when the mag fusion starts, I don't see why we should have one for GPS yaw. The status is indeed captured in the status flags so it's a bit superfluous.